lms/djangoapps/teams/api.py

```diff
@@ -181,6 +181,34 @@ def user_organization_protection_status(user, course_key):
 
 
 def has_specific_team_access(user, team):
+    """
+    To have access to a team a user must:
+        - Be course staff
+        OR
+        - be in the correct bubble
+        - be in the team if it is private
+    """
+    return has_course_staff_privileges(user, team.course_id) or (
+        user_protection_status_matches_team(user, team) and team_is_public_or_user_on_team(user, team)
+    )
+
+
+def team_is_public_or_user_on_team(user, team):
+    """
+    The only users who should be able to see private_managed teams
+    or recieve any information about them at all from the API are:
+    - Course staff
+    - Users who are enrolled in a team in a private_managed teamset
+    * They should only be able to see their own team, no other teams.
+    """
+    course_module = modulestore().get_course(team.course_id)
+    teamset = course_module.teams_configuration.teamsets_by_id[team.topic_id]
+    if teamset.teamset_type != TeamsetType.private_managed:
+        return True
+    return CourseTeamMembership.is_user_on_team(user, team)
```

Comment:
`CourseTeamMembership.is_user_on_team(user, team)` is a much cheaper operation than loading the the course from the module store.  Can you make this the first thing we check, like so:
```python
if CourseTeamMembership.is_user_on_team(user, team):
    return True
course_module = modulestore().get_course(team.course_id)
teamset = course_module.teams_configuration.teamsets_by_id[team.topic_id]
return teamset.teamset_type != TeamsetType.private_managed
```
...and then change the function name to reflect the order of operations?

Reply:
good idea!